src/dmd/dinterpret.d

```diff
@@ -836,6 +1012,53 @@ private Expression interpret(FuncDeclaration fd, InterState* istate, Expressions
         }
         eargs[i] = earg;
     }
+ 
```

Comment:
The following large chunks of code should be factored out into separate functions.